### PR TITLE
Refactoring NewNavigation Part 2

### PR DESCRIPTION
**Pull request by NataliaLKB**
## What does this change?
Refactoring the new header Navigation. [Part 1](https://github.com/guardian/frontend/pull/17568)

This change should change very little in terms of what is displayed for a user. Any changes or simplifications will happen in a separate PR.

cc @stephanfowler @zeftilldeath 

## What is the value of this and can you measure success?
Personally I find this more readable and easy to reason about when its organised in this way.

## Does this affect other platforms - Amp, Apps, etc?
Nope 

## Does this affect GLabs Paid Content Pages? Should it have support for Paid Content?
Nope

## Screenshots
No visual difference 

## Tested in CODE?
- [ ] coming soon....

### CODE REVIEW

File: common/app/navigation/NewNavigation.scala [hunk 1]
```diff
@@ -37,173 +32,6 @@ object NewNavigation {
...
-      List(lifestyle, fashion, food, recipes, loveAndSex, home),
-      List(health, family, travel, money)
-    )
-    val int = NavLinkLists(
-      List(lifestyle, fashion, food, recipes, loveAndSex, health),
-      List(home, women, family, travel, money)
-    )
-  }
-
   case object BrandExtensions extends EditionalisedNavigationSection {
```
Comment (NataliaLKB) [inline common/app/navigation/NewNavigation.scala]: I will work on refactoring what is left in this file in a separate PR

File: static/src/stylesheets/layout/new-header/_subnav.scss [hunk 2]
```diff
@@ -7,7 +7,6 @@
 
     @include mq(mobile) {
         max-height: $gs-baseline * 3.5;
-        max-height: 13vw;
```
Comment (NataliaLKB) [inline static/src/stylesheets/layout/new-header/_subnav.scss]: This didn't seem necessary to me. But maybe @gustavpursche @zeftilldeath can let me know if it is

File: common/app/navigation/GlobalNavigation.scala [hunk 3]
```diff
@@ -0,0 +1,46 @@
+package navigation
+
+object GlobalNavigation {
```
Comment (gustavpursche) [inline common/app/navigation/GlobalNavigation.scala]: Why not calling it Pillars? The rest of the structure is way more readable 👍

**Review by gu-stav — COMMENTED** [review]
This is huge! 💥 Thanks for doing that! Can't say much about the Scala code, but to me the structure looks much cleaner and understandable.

File: common/app/navigation/GlobalNavigation.scala (hunk 3 above)
Comment (NataliaLKB) [inline common/app/navigation/GlobalNavigation.scala]: Good suggestion! I will rename it

File: common/app/navigation/helpers/helpers.scala [hunk 4]
```diff
@@ -48,42 +46,30 @@ object NavigationHelpers {
...
-    val activeSectionLink = getSectionLink(sectionOrTagId)
+    val subNav = sectionLinks.find(_.id == sectionOrTagId)
 
-    (sectionOrTagId, activeSectionLink.getOrElse(""))
+    subNav
   }
 
-  private def getSectionLink(id: String): Option[String] = {
-    sectionLinks.find(_.pageId == id).map(_.parentSection.name)
+  def getSectionsToDisplay(item: NavLink2, edition: Edition): List[NavLink2] = {
```
Comment (NataliaLKB) [inline common/app/navigation/helpers/helpers.scala]: I really dislike all of this. Maybe I could do something like this instead:
```
    item.children.map(_.getEditionalisedList(edition))
        .getOrElse(item.parent.map(_.children.map(_.getEditionalisedList(edition)))
        .getOrElse(item.pillar.children.getEditionalisedList(edition)))
```

Comment (NataliaLKB) [inline common/app/navigation/helpers/helpers.scala]: But that is also really hard to read

Comment (NataliaLKB) [inline common/app/navigation/helpers/helpers.scala]: I have updated with something maybe better.

**Comment by NataliaLKB** [comment]
Closing as there are more ways to simplify this. I will do those and then reopen this.

**Pull request closed without merging**